packages/wrangler/src/index.tsx

```diff
@@ -677,7 +678,34 @@ function createCLIParser(argv: string[]) {
 				sendMetrics: config.send_metrics,
 			});
 
-			const scriptName = getLegacyScriptName(args, config);
+			let scriptName;
+
+			// Worker names can't contain "." (and most routes should), so use that as a discriminator
+			if (args.name?.includes(".")) {
+				const zone = await getZoneForRoute(args.name);
+				if (!zone) {
+					throw new Error(`Could not find zone for route ${args.name}`);
+				}
+				const routes = await getRoutesForZone(zone.id);
+
+				// Find an exact match for the route provided
+				for (const route of routes) {
+					if (route.pattern === args.name) {
+						scriptName = route.script;
+						if (args.format === "pretty") {
+							logger.log(
+								`Connecting to worker ${scriptName} at route ${args.name}`
```

Comment:
does tail already say what worker we're connecting to?

Reply:
Yes, but later in the process -- if the connection fails, we won't know what we're connecting to.

Reply:
So maybe it's better to add a logger statement earlier in the tail lifecycle, before anything can fail?